### DISCUSSION TITLE
Livewire::test() does not resolve Model Binding

### DISCUSSION THREAD

**Discussion by lairg99** [livewire/livewire#4564] [1. Help] [answered]
Hi there,

assuming we have a simple component for creating a `User`.

```php
class CreateUser extends Component {
	public User $user;

	public function mount(User $user) {
		$this->user = $user;
	}
}
```

The component is initiated by user id. For example:

```blade
<livewire:create-user :user="1"/>
```

This works fine as aspected. But when trying to test this component, the Model won't be resolved by id.

```php
Livewire::test(CreateUser::class, ['user' => 1]); 
// Throws an exception:
// ViewException : Cannot assign string to property ...\CreateUser::$template of type ...\User (View: ...)
```

Is this an expected behaviour?

**Comment by jonneroelofs** [comment]
Not sure if it is expected or not :). But it is route model binding that is not working and you are not actually going to a route, so it sort of makes sense I guess.

But what you can do is pass the model in your test instead of the id. 
```php
Livewire::test(CreateUser::class, ['user' => $user]); 
```

**Reply by lairg99** [reply]
Thank you!

**Comment by joshhanley** [comment, marked as answer]
@lairg99 yeah @jonneroelofs is correct, I would be passing the full user model into the test.

If anything doing this `<livewire:create-user :user="1"/>` shouldn't work, but as Livewire supports route model binding (where the ID is normally passed from the route), then it's by chance that this works.

But I wouldn't rely on that functionality, and instead just pass the user into your component `<livewire:create-user :user="$user"/>`

Hope this helps!

**Reply by lairg99** [reply]
Thank you! Then I'll pass the user model into the test. 
 Actually, I don't initiate the component via tag, but as an event to open a modal. So I'm forced to pass only the ID.
